src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,441 @@
+     * @throws IllegalAccessException
+     * @throws InstantiationException
+     */
+    @Override
+    public Object getInstance(Class<?> pluggableClass) throws IllegalAccessException, InstantiationException {
+        Annotation annot = null;
+        final String simpleName = pluggableClass.getSimpleName();
+
+        if (allDiscoveredAnnotations.containsKey(simpleName)) {
+            // we found a plugin class with a name that collides with an existing class;
+            // plugin names must be unique even across packages
+            throw new IllegalArgumentException(
+                    String.format("A plugin class name collision was detected (%s/%s). " +
+                                    "Simple names of plugin classes must be unique across packages.",
+                            pluggableClass.getName(),
+                            allDiscoveredAnnotations.get(simpleName).getClass().getName())
+            );
+        } else if (toolDefaultAnnotations.containsKey(simpleName)) {
+            // an instance of this class was provided by the tool as one of it's default annotations;
+            // use the default instance as the target for command line argument values
+            // rather than creating a new one, in case it has state provided by the tool
+            annot = toolDefaultAnnotations.get(simpleName);
+        } else {
+            annot = (Annotation) pluggableClass.newInstance();
+        }
+
+        // Add all annotations to the allDiscoveredAnnotations list, even if the instance came from the
+        // tool defaults list (we want the actual instances to be shared to preserve state)
+        allDiscoveredAnnotations.put(simpleName, annot);
+
+        populateAnnotationGroups(simpleName, annot);
+
+        return annot;
+    }
+
+    // Dynamic discovery of annotation groups
+    // We must discover annotation groups and store them for each instance so we can resolve group membership
+    // for command line including of groups based on their simple name.
+    private void populateAnnotationGroups(final String simpleName, final Annotation annot) {
+        Queue<Class<?>> interfaces = new LinkedList<>();
+        Collections.addAll(interfaces, annot.getClass().getInterfaces());
+        while (!interfaces.isEmpty()) {
+            Class<?> inter = interfaces.poll();
+            // Following with how groups are currently defined and discovered, namely they are interfaces that
+            // extend Annotation, groups are discovered by interrogating annotations for their interfaces and
+            // associating the discovered annotations with their defined groups.
+            if ((inter != pluginBaseClass) && (pluginBaseClass.isAssignableFrom(inter))) {
+                discoveredGroups.merge(inter.getSimpleName(), Collections.singletonList(annot), (a, b) -> Stream.concat(a.stream(), b.stream()).collect(Collectors.toList()));
+                Collections.addAll(interfaces, inter.getInterfaces());
```

Comment:
Nice catch, this is actually a pretty bad oversight. I think the best solution is probably to check in the method and not overwrite (as that probably indicates a collision and the interesting ones come first)

Reply:
As it turns out, this would have been okay for other reasons. Ultimately the getInstances() method ensured that if an instance of an annotation was passed in as a tool default it would be taken over instantiating a newer version, thus every duplicated copy would end up being a pointer to the same object, which would ultimately would be collapsed in the end by the tree set the results are returned by. I fixed it and added a test that enforces getInstanceBehavior.